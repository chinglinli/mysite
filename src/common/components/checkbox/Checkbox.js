import React from 'react'
import PropTypes from 'prop-types'

import styles from './Checkbox.scss'

export default class Checkbox extends React.PureComponent {
  static PropTypes = {
    checked: PropTypes.bool.isRequired,
    setChecked: PropTypes.func.isRequired
  }

  onChange = (e) => {
    this.props.setChecked(e.target.checked)
  }

  render () {
    return (
      <label className={styles.root}>
        <input type="checkbox" checked={this.props.checked} onChange={this.onChange}/>
        <div className={styles.box}/>
      </label>
    )
  }
}
