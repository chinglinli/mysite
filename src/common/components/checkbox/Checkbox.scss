@import '../../css/base.scss';

.root {
  display: block;
  width: 1.5rem;
  height: 1.5rem;

  input {
    display: none;
    &:checked + .box {
      &:after {
        top: 0;
      }
    }
  }

  .box {
    width: 100%;
    height: 100%;
    transition: all 1.1s cubic-bezier(.19, 1, .22, 1);
    border: 1px solid #ccc;
    background-color: #f9f9f9;
    position: relative;
    overflow: hidden;
    cursor: pointer;

    &:after {
      width: 50%;
      height: 20%;
      content: '';
      position: absolute;
      border-left: 0.3rem solid;
      border-bottom: 0.3rem solid;
      border-color: $accent-color;
      transform: rotate(-45deg) translate3d(0, 0, 0);
      transform-origin: center center;
      transition: all 1.1s cubic-bezier(.19, 1, .22, 1);
      left: 0;
      right: 0;
      top: 200%;
      bottom: 5%;
      margin: auto;
    }
  }
}
