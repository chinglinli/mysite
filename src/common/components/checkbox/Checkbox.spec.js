import React from 'react'
import renderer from 'react-test-renderer'

import Checkbox from './Checkbox'

test('It should render Checkbox component in checked state correctly', () => {
  const component = renderer.create(
    <Checkbox checked={false} setChecked={() => null}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should render Checkbox component in checked state correctly', () => {
  const component = renderer.create(
    <Checkbox checked={true} setChecked={() => null}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
