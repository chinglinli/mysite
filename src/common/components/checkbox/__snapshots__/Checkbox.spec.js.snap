// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Checkbox component in checked state correctly 1`] = `
<label
  className="root"
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="box"
  />
</label>
`;

exports[`It should render Checkbox component in checked state correctly 2`] = `
<label
  className="root"
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="box"
  />
</label>
`;
